recipes/openeo/meta.yaml

```diff
@@ -0,0 +1,70 @@
+{% set name = "openeo" %}
+{% set version = "0.8.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/Open-EO/openeo-python-client/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 1fe6339a960216bb61083f356bdc94063337df241253fff723f9006ac1fa20a2
+
+build:
+  number: 0
+  #skip: True  # [not win]" to limit to Windows.
+  skip: True  # [py<35]
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
+    - pip
+    - setuptools
+    - deprecated
+    - geopandas
+    - graphviz
+    - h5netcdf
+    - matplotlib
+    - numpy
+    - pandas
+    - pytest-runner
+    - requests
+    - shapely
+    - xarray
+  run:
```

Comment:
where does this `run` requirements listing come from? The basic install requirements are currently (https://github.com/Open-EO/openeo-python-client/blob/547e5b7d3c9f6327915228c530cd7ffef3b53eb8/setup.py#L41-L49):
```
          'requests',
          'shapely>=1.6.4',
          'numpy>=1.17.0',
          'xarray>=0.12.3',
          'pandas>0.20.0;python_version>="3.5.3"',
          'pandas<0.25.0;python_version<"3.5.3"',
          'deprecated>=1.2.12'
```

For example, I don't think it is necessary to force users to install  matplotlib, geoandas, graphviz

Reply:
more general note (as I don't know conda-forge at the moment): isn't there a way to use the dependencies defined in the openeo package directly (setup.py), instead of redefining them in meta.yaml?

Reply:
Unfortunately, they need to be explicitly listed in the `meta.yaml`.

Reply:
Hi,

I used https://github.com/conda-forge/conda_deps-feedstock to gather the requirements directly from the source code but the output is certainly not accurate since it just gathers all `import` statements from the Python files. Thanks for clarifying what are the real requirements. I have now updated the recipe and I have also updated the `openeo-python-client` version to the last release (`v0.8.2`). Let's see whether all tests pass now.

Reply:
looks better now, thanks.
Any reason `h5netcdf` and `matplotlib-base` are still listed?

Reply:
Thanks.
I am sure `pytest` failed without `matplotlib-base` so I moved it to the requirements section of the tests.
I have now deleted `h5netcdf`. Let's see the outcome of the tests without it.

Reply:
This is one of the several errors that I get without `h5netcdf`:
```
_____________________ test_run_local_udf_from_file_netcdf ______________________

tmp_path = PosixPath('/tmp/pytest-of-conda/pytest-0/test_run_local_udf_from_file_n0')

        if isinstance(filename, bytes):
            raise ValueError(
                "can't open netCDF4/HDF5 as bytes "
                "try passing a path or file-like object"
            )
        elif isinstance(filename, io.IOBase):
            magic_number = read_magic_number_from_file(filename)
            if not magic_number.startswith(b"\211HDF\r\n\032\n"):
                raise ValueError(
                    f"{magic_number} is not the signature of a valid netCDF4 file"
                )
    
        if format not in [None, "NETCDF4"]:
            raise ValueError("invalid format for h5netcdf backend")
    
        kwargs = {"invalid_netcdf": invalid_netcdf}
        if phony_dims is not None:
            if LooseVersion(h5netcdf.__version__) >= LooseVersion("0.8.0"):
                kwargs["phony_dims"] = phony_dims
            else:
                raise ValueError(
                    "h5netcdf backend keyword argument 'phony_dims' needs "
                    "h5netcdf >= 0.8.0."
                )
>       if LooseVersion(h5netcdf.__version__) >= LooseVersion(
            "0.10.0"
        ) and LooseVersion(h5netcdf.core.h5py.__version__) >= LooseVersion("3.0.0"):
E       NameError: name 'h5netcdf' is not defined

../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/xarray/backends/h5netcdf_.py:166: NameError
```

Reply:
It looks like `h5netcdf` is used as an `xarray` backend:
```
lib/python3.9/site-packages/xarray/backends/h5netcdf_.py:166: NameError
```
@soxofaan should I add `h5netcdf` back to the `run` requirements?

Reply:
sorry for the delay

how do you trigger that h5netcdf error? Do you have a code snippet to reproduce that?

Reply:
No worries.
That error is produced when I remove `h5netcdf` from the `run` requirements from the conda recipe. See [this](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=370539&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=2016) example.

Reply:
but that error is during the tests
can you add h5netcdf as a test dependency? 
see https://github.com/Open-EO/openeo-python-client/blob/3d62accf61c7c2db017879cace55fb32381a8c32/setup.py#L20-L27

Reply:
Sorry, I missed that.
I have now harmonized the requirements in the conda recipe.
Could you please double check and let me know your thoughts?
Do you have interest in having this working on Windows or should we disable the build for it?

Reply:
the package listings now look ok to me

> Do you have interest in having this working on Windows
yes, proper windows support is quite imporant for this package

Reply:
Thanks @soxofaan 
I am now disabling Windows builds and will ping the `conda-forge` team to review this PR.